### PR TITLE
Many updates

### PR DESCRIPTION
**Pull request by joaquimg**
should do the work of
#32 #33 #41 and #44 

cc @jd-lara you might want to test this one!

- Mimic most of JuMP's variable interface
- basic fixes in operator algebra
- add deps
- improve printing
- test expressions

Should also solve #43 and #39

### CODE REVIEW

**Comment by jd-lara** [comment]
This branch is working for PowerSimulations.jl. This gets 2/3 functionalities we need to work.

**Comment by joaquimg** [comment]
Its #43 what is missing?

File: test/tests.jl [hunk 1]
```diff
@@ -330,17 +330,114 @@ function test13(args...)
...
+end
+
+function test15(args...)
+    @testset "Test add ctr - alternative" begin
+        model = ModelWithParams(args...)
+        α = Parameter(model, 1.0)
+        fix(α, -1.0)
+        @variable(model, x)
+        exp = x - α
+        cref = @constraint(model, exp ≤ 0)
```
Comment (joaquimg) [inline test/tests.jl:436]: @jd-lara creating constraint from expression looks good!

File: src/constraints.jl [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+const GAEp{C} = JuMP.GenericAffExpr{C,Parameter}
+
+mutable struct DoubleGenericAffExpr{C,V,P} <: JuMP.AbstractJuMPScalar
+    v::JuMP.GenericAffExpr{C,V}
+    p::JuMP.GenericAffExpr{C,P}
+end
+const DGAE{C,V,P} = DoubleGenericAffExpr{C,V,P}
+
+const ParametrizedGenericAffExpr{C,V} = DoubleGenericAffExpr{C, V, Parameter}
+const PGAE{C,V} = ParametrizedGenericAffExpr{C,V}
```
Comment (joaquimg) [inline src/constraints.jl:15]: @jd-lara you might wanna try this type!

Comment (jd-lara) [reply src/constraints.jl:15]: I will update PowerSimulations to use this now in the ExpressionArrays

Comment (jd-lara) [reply src/constraints.jl:15]: This is working pretty well!

**Comment by joaquimg** [comment]
@jd-lara I am planning to merge today, do you still have any suggestions?

**Comment by jd-lara** [comment]
@joaquimg I haven't run into any other issues in the model building phase. You can merge anytime. 

Probably more stuff will come up when developing the sequential model solving.

**Pull request merged**